java/arcs/core/data/expression/Expression.kt

```diff
@@ -360,11 +360,11 @@ sealed class Expression<T> {
      * @param E the type of the [qualifier] expression if any
      * @param T the type of elements in the resulting [Sequence]
      */
-    data class FromExpression<E, T>(
-        override val qualifier: Expression<Sequence<E>>?,
-        val expr: Expression<Sequence<T>>,
+    data class FromExpression(
+        override val qualifier: Expression<Sequence<Unit>>?,
+        val source: Expression<Sequence<Any>>,
         val iterationVar: String
-    ) : QualifiedExpression<E>, Expression<Sequence<T>>() {
+    ) : QualifiedExpression<Unit>, Expression<Sequence<Unit>>() {
```

Comment:
The usage of Unit does look a little strange, eventually we'll have to eliminate the args I think.

Reply:
I've turned the QualifiedExpression into `Expression<Sequence<Unit>>`, I don't think it every makes sense to use qualified expression (WHERE, SELECT) on sequences other than created by FROM (as without from you don't get the scope variable), so it simplifies a bit further.